src/pybind/mgr/dashboard/services/auth/oauth2.py

```diff
@@ -6,15 +6,14 @@
 import requests
 
 from ... import mgr
-from ...services.auth import BaseAuth, SSOAuth, decode_jwt_segment
+from ...services.auth import BaseAuth, SSOAuthMixin, decode_jwt_segment
 from ...tools import prepare_url_prefix
 from ..access_control import Role, User, UserAlreadyExists
 
 
-class OAuth2(SSOAuth):
+class OAuth2(SSOAuthMixin):
```

Comment:
My suggestion was more (this avoid "diamond inheritance", which is not really an issue in Python, but not really required as `SSOAuthMixin` doesn't implement anything from `BaseAuth`):
```suggestion
class OAuth2(BaseAuth, SSOAuthMixin):
```

Reply:
changed it